spec/lib/phobos/executor_spec.rb

```diff
@@ -81,4 +106,33 @@ class TestHandler2 < Phobos::EchoHandler; end
       wait_for_event('executor.stop')
     end
   end
+
+  context 'custom exponential backoff' do
```

Comment:
Hmm, I am missing a negative test here (if you could call it that). We are only testing the custom. This might be a bit overwhelming but I have a few requests so bear with me, I don't think it's that much just a lot of text:

1. I would suggest to create a context backoff testing both default and custom override. Pseudo:

```ruby
context 'backoff' do
  it 'uses the default backoff settings specified in Phobos.config.backoff'
  context 'with override for a specific listener'
    it 'uses the custom backoff settings'
```

2. It would also be good to keep the concept of `listeners` (plural) and check the value of both of them, i e use `let(:listeners) { [...] }` and use at least two listeners. I think the easiest is to crash listener 1 in one test, and listener 2 in a separate test since otherwise you might fight too much with concurrency issues.

3. Please try to follow the convention of subscribing to events and waiting for them, it makes the file cleaner and consistent, and you could probably remove your `before` block which seems a bit "doing what needs to be done to make it pass" (it actually isn't really obvious what its doing or why its needed). 

4. The test is testing the wrong thing Im afraid 💀  The problem is that you are actually mocking the value - not asserting it. Never use expect().and_return - that would be asserting that rspec mocks is working. 

To verify this I tested to change the value of `min_ms: 3748` and it still passed. I tried removing the whole `backoff: { .. }` - it passed. I tried `expect_any_instance_of(ExponentialBackoff).to receive(:interval_at).once.and_return(3.0)` and... it passed. 

So... I know this isn't pretty and it is kinda hacky, but it actually asserts the correct stuff and would give reliable errors in case things breaks:

```ruby
      expect_any_instance_of(ExponentialBackoff).to receive(:interval_at) do |backoff|
        expect(backoff.instance_variable_get(:@maximum_elapsed_time)).to eq(5.678)
        expect(backoff.instance_variable_get(:@minimal_interval)).to eq(3.748)
        1
      end
```

And I use values that are a little bit more characteristic, a wild "2" or "3" might appear anywhere so I prefer more unique values. Using this way you can in fact remove the line `allow_any_instance_of(ExponentialBackoff).to receive(:randomize_factor=)` since here we are asserting before returning the value 1.

Reply:
@klippx Thanks for your excellent feedback and attention to detail!

I was having some trouble understanding how everything in the test file worked at first. After reviewing the file and your feedback it became apparent how the test _should_ be written. I'm also more familiar with the executor now 👍 

Third time is the charm?